Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -107,6 +127,45 @@ public void Razor_ExcludedProject_NoErrorLog()
             AssertExpectedErrorLog(result, string.Empty);
         }
 
+        [TestMethod]
+        public void Razor_RazorSpecificOutputAndProjectInfo_AreCopiedToCorrectFolders()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "0");
+            var temporaryProjectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, @"0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.Razor");
+            var protoBufFile = Path.Combine(temporaryProjectSpecificOutDir, "ProtoBuf.txt");
+            var issues = Path.Combine(projectSpecificOutDir, "Issues.json");
+            File.WriteAllText(protoBufFile, string.Empty);
+            File.WriteAllText(issues, string.Empty);
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <SonarTemporaryProjectSpecificOutDir>{temporaryProjectSpecificOutDir}</SonarTemporaryProjectSpecificOutDir>
+  <ProjectSpecificOutDir>{projectSpecificOutDir}</ProjectSpecificOutDir>
+  <RazorSonarErrorLog>{issues}</RazorSonarErrorLog>
+</PropertyGroup>
+
+<ItemGroup>
+  <RazorCompile Include='SomeRandomValue'>
+  </RazorCompile>
+</ItemGroup>
+";
+
+            var filePath = CreateProjectFile(null, projectSnippet, TargetConstants.SonarFinishRazorCodeAnalysis);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.SonarFinishRazorCodeAnalysis);
+
+            // Assert
+            var actualProjectInfo = ProjectInfoAssertions.AssertProjectInfoExists(root, Path.Combine(root, "Razor_RazorSpecificOutputAndProjectInfo_AreCopiedToCorrectFolders.proj.txt"));
+            result.AssertTargetExecuted(TargetConstants.SonarFinishRazorCodeAnalysis);
+            File.Exists(Path.Combine(projectSpecificOutDir, "ProtoBuf.txt")).Should().BeTrue();
+            actualProjectInfo.AnalysisSettings.FirstOrDefault(x => x.Id.Equals("sonar.cs.analyzer.projectOutPaths")).Value.Should().Be(razorSpecificOutDir);
```

Comment:
Will there be more or default?
```suggestion
            actualProjectInfo.AnalysisSettings.Single(x => x.Id.Equals("sonar.cs.analyzer.projectOutPaths")).Value.Should().Be(razorSpecificOutDir);
```